📚 Docs: Clarify that `auth.uid()` returns `null` when unauthenticated in RLS policies (#37336)
supabase/supabase: apps/docs/content/guides/database/postgres/row-level-security.mdx

Once you have enabled RLS, no data will be accessible via the [API](/docs/guides/api) when using the public `anon` key, until you create policies.

> ### ⚠️ Be Aware: `auth.uid()` Returns `null` When Unauthenticated

Reviewer: ```suggestion
### ⚠️ Be Aware: `auth.uid()` Returns `null` When Unauthenticated
<Admonition type="warning" label="`auth.uid()` Returns `null` When Unauthenticated">
```

> When a request is made without an authenticated user (e.g., no access token is provided or the session has expired), `auth.uid()` will return `null`.

Reviewer: ```suggestion
When a request is made without an authenticated user (e.g., no access token is provided or the session has expired), `auth.uid()` returns `null`.
```

This means that a policy like:

```sql
USING (auth.uid() = user_id)
```
will silently fail for unauthenticated users, because:
```sql
null = user_id
```
is always false in SQL.

To avoid confusion and make your intention clear, we recommend explicitly checking for authentication:
```sql
USING (auth.uid() IS NOT NULL AND auth.uid() = user_id)
```

> 

Reviewer: ```suggestion
</Admonition>
```
